application.go

```diff
@@ -431,12 +429,14 @@ func (r *marathonClient) WaitOnApplication(name string, timeout time.Duration) e
 			flick.SwitchOn()
 		}()
 		for !flick.IsSwitched() {
-			if found, err := r.HasApplication(name); err != nil {
-				continue
-			} else if found {
-				if app, err := r.Application(name); err == nil && app.AllTaskRunning() {
-					return nil
+			if err := r.HasApplication(name); err != nil {
```

Comment:
I think `HasApplication` isn't really needed here: You can call `Application` and return values depending on the error received.

This is even more true if we agree on removing `HasApplication` in its entirety.